### DISCUSSION TITLE
Header & Footer Files

### DISCUSSION THREAD

**Discussion by jonathan-wondereur** [gotenberg/gotenberg#430] [Q&A]
It is unclear to me where to put the files for the header and footer based on the documentation. I have been unable to get the files to load in my output using V7

**Comment by gulien** [comment]
Hello @jonathan-wondereur 

This feature is only for conversions with the Chromium module. You have to send files named `header.html` and `footer.html` in your form data and Gotenberg will automatically find them.

For instance:

```
curl \
--request POST 'http://localhost:3000/forms/chromium/convert/url' \
--form 'url="https://my.url"' \
--form 'files=@"/path/to/header.html"' \
--form 'files=@"/path/to/footer.html"' \
--form 'marginTop="1"' \
--form 'marginBottom="1"' \
-o my.pdf 
```

**Reply by gulien** [reply]
I’m curious which part of the documentation is not clear btw 😄

https://gotenberg.dev/docs/modules/chromium#routes -> there is a dedicated tab for the Header/Footer feature

**Reply by jonathan-wondereur** [reply]
Sure, it works with the curl example, but how exactly am I supposed to send the file? I am making a post request from Ruby on Rails and I do not know how I am supposed to send the file? From the documentation, it seems like the file should be on the local file system of the Gotenberg instance. Is that not the case? If not, how, and in what format should I send the file using a Post request from Ruby on Rails?

**Reply by jonathan-wondereur** [reply]
Okay, I found an example of how to post a file (https://github.com/jnunemaker/httparty/blob/master/examples/multipart.rb). what is unclear about the documentation is that I need to send a file (as I am not familiar with CURL syntax for this).

**Reply by gulien** [reply]
Actually they are form files, meaning you can "hydrate" them as you want 😄 i.e., either from a file, or a string, or something else.

For instance, the PHP client takes a `Stream` class:

* https://github.com/gotenberg/gotenberg-php/blob/v1.0.0/src/Modules/Chromium.php#L114
* https://github.com/gotenberg/gotenberg-php/blob/v1.0.0/src/Stream.php
